zerver/views/users.py

```diff
@@ -386,9 +368,43 @@ def avatar(
             # If there is a valid user account passed in, use its avatar
             url = avatar_url(avatar_user_profile, medium=medium)
         assert url is not None
+    except UserProfile.DoesNotExist:
+        url = get_avatar_for_inaccessible_user()
```

Comment:
In this case, we can't fall back to the Gravatar URL. Is this ok, or would we prefer to call something else? Previously it'd fail inside `get_gravatar_url`.

```python
except UserProfile.DoesNotExist:
        # If there is no such user, treat it as a new gravatar
        email = email_or_id
        avatar_version = 1
        url = get_gravatar_url(email, avatar_version, medium)
```

Reply:
Yeah this seems like a correct bug fix to the previous logic.